jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/CheckForDuplicatePaths.java

```diff
@@ -0,0 +1,160 @@
+package se.fortnox.reactivewizard.jaxrs;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import se.fortnox.reactivewizard.util.ReflectionUtil;
+
+import javax.inject.Inject;
+import javax.ws.rs.CookieParam;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Type;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CheckForDuplicatePaths extends StartupCheck {
+
+    private final        StartupCheckConfig startupCheckConfig;
+    private static final Logger             LOG                = LoggerFactory.getLogger(CheckForDuplicatePaths.class);
+    public static final  Pattern            PATH_PARAM_PATTERN = Pattern.compile("\\{(\\w*)\\}");
+
+    @Inject
+    public CheckForDuplicatePaths(StartupCheckConfig startupCheckConfig, JaxRsRequestHandler jaxRsRequestHandler) {
+        this.startupCheckConfig = startupCheckConfig;
+        final List<JaxRsResource> resources = jaxRsRequestHandler.getResources().getResources();
+
+        check(resources);
+    }
+
+    void check(List<JaxRsResource> resources) {
+        for (int i = 0; i < resources.size() - 1; i++) {
+            JaxRsResource testedResource = resources.get(i);
+            for (int j = i + 1; j < resources.size(); j++) {
+                JaxRsResource otherResource = resources.get(j);
+                if (pathParamCollides(testedResource, otherResource)) {
+                    String errorMessage = String.format("%s duplicates %s on path %s",
+                        testedResource.getResourceMethod(),
+                        otherResource.getResourceMethod(),
+                        testedResource.getPath()
+                    );
+                    if (startupCheckConfig.isFailOnError()) {
+                        throw new IllegalStateException(errorMessage);
+                    } else {
+                        LOG.warn(errorMessage);
+                    }
+                }
+            }
+        }
+    }
+
+    public static boolean pathParamCollides(JaxRsResource jaxRsResource1, JaxRsResource jaxRsResource2) {
+        if (jaxRsResource2 == null) {
```

Comment:
this if statement should not be needed? we should never be able to get into this state if the loop works as intended?